runtime/lua/vim/lsp/buf.lua

```diff
@@ -15,12 +15,13 @@ local function npcall(fn, ...)
   return ok_or_nil(pcall(fn, ...))
 end
 
-local function request(method, params, callback)
+local function request(method, params, callback, unsupported_callback)
   validate {
     method = {method, 's'};
     callback = {callback, 'f', true};
+    unsupported_callback = {unsupported_callback, 'f', true};
   }
-  return vim.lsp.buf_request(0, method, params, callback)
+  return vim.lsp.buf_request(0, method, params, callback, unsupported_callback)
```

Comment:
I think you also need to change `vim.lsp.buf_request` to accept `unsupported_callback`. As far as I can tell right now it is not used at all (just passes an extra arg to that function).

Also needs doc updates to add it.

Reply:
I think this was left-over from a previous approach. Removed.